src/main/java/io/appium/java_client/HasSessionDetails.java

```diff
@@ -34,26 +35,46 @@
     @SuppressWarnings("unchecked")
     default Map<String, Object> getSessionDetails() {
         Response response = execute(GET_SESSION);
+        Map<String, Object> resultMap = Map.class.cast(response.getValue());
+
         return  ImmutableMap.<String, Object>builder()
```

Comment:
It might be helpful to have a comment here about why additional filtering is necessary